eng/readytorun.targets

```
@@ -0,0 +1,36 @@
+<Project>
+  <PropertyGroup>
+    <!-- Only use mibc files if UsingToolIbcOptimization is false. Allows enabling/disabling using ibc instead of mibc data -->
+    <IncludePgoFiles Condition="'$(UsingToolIbcOptimization)' != 'true' and '$(EnableNgenOptimization)' == 'true' and '$(DotNetBuildFromSource)' != 'true'">true</IncludePgoFiles>
+  </PropertyGroup>
+
+  <Target Name="ResolveReadyToRunCompilers" DependsOnTargets="ResolveRuntimeFilesFromLocalBuild">
+    <ItemGroup>
+      <OptimizationMibcFiles Include="$(CoreCLRArtifactsPath)StandardOptimizationData.mibc" Condition="'$(DotNetBuildFromSource)' != 'true'" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <PublishReadyToRunPgoFiles Condition="'$(IncludePgoFiles)' == 'true'" Include="$(CoreCLRArtifactsPath)StandardOptimizationData.mibc"/>
+      <PublishReadyToRunCrossgen2ExtraArgsList Condition="'$(IncludePgoFiles)' == 'true'" Include="--embed-pgo-data"/>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <Crossgen2Dll>$(CoreCLRArtifactsPath)\$(BuildArchitecture)\crossgen2\crossgen2.dll</Crossgen2Dll>
```

NIT/Question: What's the criteria to define whether we write paths with backslashes, forward slashes, or .NET's `Path` API? I think it would be best to gradually standardize it for maximum safety. While I prefer forward slashes, I believe using the `Path` API would be best to avoid potential compatibility pitfalls.

---

In this case, it would be better to use MSBuild's normalization function though as it automatically takes care of using the right directory separators (which makes the path look correct in a binlog).

```suggestion
      <Crossgen2Dll>$([MSBuild]::NormalizePath('$(CoreCLRArtifactsPath)', '$(BuildArchitecture)', 'crossgen2', 'crossgen2.dll'))</Crossgen2Dll>
```